lib/game_client/ogs/ogs_game_client.dart

```diff
@@ -153,15 +252,114 @@ class OGSGameClient extends GameClient {
 
   @override
   Future<Game> findGame(String presetId) async {
-    throw UnimplementedError();
+    // Parse the preset ID to extract board size and speed (format: "9_blitz", "19_live", etc.)
+    final parts = presetId.split('_');
+    final boardSize = int.parse(parts[0]);
+    final speed = parts.length >= 2 ? parts[1] : 'rapid';
+
+    // Generate a UUID for the automatch request and store it for later cancellation
+    _currentAutomatchUuid = const Uuid().v4();
+    _automatchCompleter = Completer<Game>();
+
+    // Set up message listener for automatch responses
+    _messageSubscription =
+        _webSocketManager.messages.listen(_handleAutomatchResponse);
+
+    // Create the automatch payload data matching OGS format
+    final automatchData = {
+      "uuid": _currentAutomatchUuid!,
+      "size_speed_options": [
+        {"size": "${boardSize}x$boardSize", "speed": speed, "system": "byoyomi"}
+      ],
+      "lower_rank_diff": 3,
```

Comment:
why specifically `3`? does this match what OGS does?

Reply:
Exactly, it's the default: https://github.com/online-go/online-go.com/blob/b0d661e69cef0ce57c2c5d4e2e04109227ba9a96/src/lib/preferences.ts#L57-L58
I added some breadcrumbs here for future readers :)